From 42ae902e8e21d6c1f469374a8bede1ac10d8a579 Mon Sep 17 00:00:00 2001
From: Long Dang <long.dang.aj@rvc.renesas.com>
Date: Thu, 18 Aug 2016 17:50:56 +0700
Subject: [PATCH 2/3] gst145-gstomx:-fix memory leak in _class_init function

Porting from a patch in Linux 3.10 environment
References: developer.gnome.org/glib/stable/glib-Key-value-file-parser

g_key_file_get_string_list() - RETURN:
"a NULL-terminated string array or NULL if the specified key cannot be found.
The array should be freed with g_strfreev()."

Signed-off-by: Long Dang <long.dang.aj@rvc.renesas.com>
---
 omx/gstomx.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/omx/gstomx.c b/omx/gstomx.c
index f530847..82e36ee 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -2649,6 +2649,7 @@ _class_init (gpointer g_class, gpointer data)
 
     class_data->hacks = gst_omx_parse_hacks (hacks);
   }
+  g_strfreev(hacks);
 }
 
 static gboolean
-- 
1.9.1

